crates/clorinde/src/codegen/cargo.rs

```diff
@@ -43,253 +40,208 @@ impl DependencyAnalysis {
-            "#}
-            .unwrap();
-            write!(client_features, r#""with-eui48-1","#).unwrap();
+        _ => {
+            deps_toml.push_str(
+                "\n\
+            chrono = []\n\
+            time = []
+            ",
+            );
         }
-        if dependency_analysis.decimal {
-            writedoc! { buf, r#"
-                # DECIMAL
-                rust_decimal = {{ version = "1.36.0", features = ["db-postgres"] }}
-            "#}
-            .unwrap();
+    };
+    deps_toml.push_str(
+        "\n\
+        [dependencies]\n\
+        ## Core dependencies\n\
+        # Postgres types\n\
+        postgres-types = {\"0.2.8\", features = [\"derive\"] }\n\
+        # Postgres interaction\n\
+        postgres-protocol = {\"0.6.7\"}\n\
+        # Iterator utils required for working with `postgres_protocol::types::ArrayValues`\n\
+        fallible-iterator = {\"0.2.0\"}
+        ",
+    );
+    match dependency_analysis.has_dependency() {
+        true => deps_toml.push_str(
+            "\n\
+            ## Types dependencies\n\
+            # TIME, DATE, TIMESTAMP or TIMESTAMPZ\n\
+            chrono = {\"0.4.39\", optional = true }
+            time = {\"0.3.37\", optional = true }
+        ",
+        ),
+        _ => {}
+    }
+    match dependency_analysis.uuid {
+        true => deps_toml.push_str(
+            "\n\
+            # UUID
+            uuid = {\"1.11.0\"}
+        ",
+        ),
+        _ => {}
+    };
+    match dependency_analysis.mac_addr {
+        true => deps_toml.push_str(
+            "\n\
+           # MAC ADDRESS\n\
+           eui48 = {\"1.1.0\", default-features = false }
+        ",
+        ),
+        _ => {}
+    };
+    match dependency_analysis.decimal {
+        true => deps_toml.push_str(
+            "\n\
+            # DECIMAL\n\
+            rust_decimal = {\"1.36.0\", features = [\"db-postgres\"] }
+        ",
+        ),
+        _ => {}
+    };
+    match dependency_analysis.uuid {
+        true => deps_toml.push_str(
+            "\n\
+            # JSON or JSONB\n\
+            serde_json = {\"1.0.134\", features = [\"raw_value\"] }\n\
+            serde = {\"1.0.217\", features = [\"derive\"] }\n\
+        ",
+        ),
+        _ => {}
+    };
+    match (config.serialize, !dependency_analysis.json) {
+        (true, true) => deps_toml.push_str(
+            "\n\
+            ## Serialize\n\
+            serde = {\"1.0.217\", features = [\"derive\"] }
+        ",
+        ),
+        _ => {}
+    };
+    match config.r#async {
+        true => deps_toml.push_str(
+            "\n\
+            ## Async client dependencies\n\
+            # Async utils\n\
+            futures = {\"0.3.31\"}\n\
+            \n\
+            ## Async features dependencies\n\
+            # Async connection pooling\n\
+            deadpool-postgres = {\"0.14.1\", optional = true }
+        ",
+        ),
+        _ => deps_toml.push_str(
```

Comment:
issue(blocking): with this as basically an `if/else` it is no longer possible to have `async` and `sync` at the same time.